src/app/create-recording/custom-recording/custom-recording.component.html

```diff
@@ -3,23 +3,31 @@
     <label for="recordingName">Name</label>
     <input id="recordingName" type="text" class="form-control" [(ngModel)]="name" name="recordingName" placeholder="Name" (keyup.enter)="submit()"/>
   </div>
-  <div class="form-group">
-    <label for="recordingDuration">Duration</label>
-    <select id="recordingDuration" [(ngModel)]="duration" name="recordingDuration" class="form-control">
-      <option value="-1">Unlimited</option>
-      <option value="5">5 seconds</option>
-      <option value="30">30 seconds</option>
-      <option value="60">60 seconds</option>
-      <option value="300">5 minutes</option>
-    </select>
+  <div class="form-group row">
+    <div class="col col-xs-6">
+      <label for="recordingDuration">Duration <span class="pficon-info" [tooltip]="durationTooltipTemplate" placement="right"></span></label>
+      <input id="recordingDuration" name="recordingDuration" type="number" class="form-control" [(ngModel)]="duration" min="0"/>
+    </div>
+    <div class="col col-xs-6">
+      <label for="unitMultiplier">Unit</label>
+      <select id="unitMultiplier" [(ngModel)]="unitMultiplier" name="unitMultiplier" class="form-control" [attr.disabled]="duration > 0 ? null : ''">
+        <option value="1">Seconds</option>
+        <option value="60">Minutes</option>
+        <option value="3600">Hours</option>
+      </select>
+    </div>
   </div>
   <div class="form-group">
-    <label for="recordingEvents">Events <span class="pficon-info" [tooltip]="tooltipTemplate" placement="right"></span></label>
+    <label for="recordingEvents">Events <span class="pficon-info" [tooltip]="eventTooltipTemplate" placement="right"></span></label>
     <input id="recordingEvents" type="text" class="form-control" [(ngModel)]="events" name="recordingEvents" placeholder="events" (keyup.enter)="submit()"/>
   </div>
   <button type="submit" class="btn btn-primary" (click)="submit()">Submit</button>
 </form>
 
-<ng-template #tooltipTemplate>
+<ng-template #durationTooltipTemplate>
+  <span>&lt;= 0 specifies an "unlimited" duration, continuous recording</span>
```

Comment:
Does this require any API change on the backend?

Reply:
No, that's already how it worked.